data/registry_model/shared.py

```diff
@@ -0,0 +1,17 @@
+import uuid
+
+from hashids import Hashids
+
+
+class SyntheticIDHandler(object):
+    def __init__(self, hash_salt=None):
+        self.hash_salt = hash_salt or str(uuid.uuid4())
+        self.hashids = Hashids(alphabet="0123456789abcdef", min_length=64, salt=self.hash_salt)
+
+    def encode(self, manifest_id, layer_index=0):
+        encoded = self.hashids.encode(manifest_id, layer_index)
+        assert len(encoded) == 64
+        return encoded
+
+    def decode(self, synthetic_v1_id):
+        return self.hashids.decode(synthetic_v1_id)
```

Comment:
```
self.hash_salt = hash_salt or str(uuid.uuid4())
```

Is there a test that verifies an `.encode()` followed by a `.decode()` with, and without, a value for `hash_salt`?

Reply:
I trust hashids to be tested and we do have a test somewhere that sends in an invalid value

Reply:
Added another explicit unit test just in case